src/opencl_sha256crypt_fmt_plug.c

```diff
@@ -290,6 +294,7 @@ static void set_key(char *key, int index)
 	else
 		first_len = len;
 	new_keys = 1;
+	bitmap_of_lens |= (1 << len);
```

Comment:
We should be able to drop `new_keys`, `lens_differ`, and `first_len`. We really only need `bitmap_of_lens` now.

Reply:
Let me clarify: I think we can drop even `new_keys` because we always end up setting `bitmap_of_lens` to non-zero if `set_keys()` was called at least once. So we can just check this one for non-zero instead. And we also reset it to 0 after we've first processed the new batch of keys in `crypt_all()`.

Reply:
> Let me clarify: I think we can drop even `new_keys` because we always end up setting `bitmap_of_lens` to non-zero if `set_keys()` was called at least once. So we can just check this one for non-zero instead. And we also reset it to 0 after we've first processed the new batch of keys in `crypt_all()`.

It is no longer the case